en/docs/references/ballerina-expression-syntax.md

```diff
@@ -1,422 +1,485 @@
+### Simple values in JSON 
 
-#### Examples
+Ballerina has a single type named `json` that can represent any JSON value. You can use `json` which is a built-in union type in Ballerina, to store values of the types `nil` (as the null value), `boolean`, `int`, `float`, `decimal`, `string`, `json[]`, and `map<json>`. 
 
-- Checking if a particular HTTP header (`content-length`) exists given the header name.
+###### Examples:
 
-  ```ballerina
-  request.hasHeader("content-length")
-  ```
-  
-- Checking if a particular HTTP header (`content-length`) exists by passing a variable (`contentLength`) containing the header name. Note the omission of the double-quotes.
+To initialize a `json` type variable, you can use the following:
+```ballerina
+    "Alex"
 
-  ```ballerina
-  request.hasHeader(contentLength)
-  ```
+    5.36
+    
+    false
+    
+    [1, false, null, "foo", {first: "Alex", last: "John"}]
+    
+    {name: "Alex", age: null, marks: {math: 90, language: 95}}
+    
+     null
+```
 
-### Reading the JSON Payload of the request
+### JSON Objects
 
-JSON is a common content type used for HTTP communication. The `request` variable’s `getJsonPayload` function can be used to read the JSON payload sent with the HTTP request. Note that the `getJsonPayload` function’s return type is `json|ClientError`.
+Ballerina represents JSON objects as maps. You can define a JSON object to hold values of any `json` type (`string`, `number`, `boolean`, `object`, `array`, or `null`). A JSON object can be declared in two ways:
+    -  Using the variable type `json`.
+    -  Using the variable type `other` and `map<json>` as the `other type`.
 
-#### Example
+###### Examples:
 
-- Reading the JSON Payload of the HTTP request.
 
-  ```ballerina
-  request.getJsonPayload()
-  ```
+To declare a JSON object, use the following:
 
-### Reading the Text Payload of the request
+```ballerina
+     {name: "apple", color: "red", price: "20.0"}    
+```
 
-Sometimes, content is sent as a string (plain text) with the HTTP request. The `request` variable’s `getTextPayload` function can be used to read the text payload sent with the HTTP request. Note that the `getTextPayload` function’s return type is `string|ClientError`.
+To create an empty JSON object, use the following:
 
-#### Example
+```ballerina
+    {}
+```
 
-- Reading the Text Payload of the HTTP request.
+To cast a `json` type variable `fruitDetail` to a `map<json>` type variable, use the following:
 
-  ```ballerina
-  check request.getTextPayload()
-  ```
+```ballerina
 
-### Reading the query parameter value
+<map<json>>fruitDetail
+ 
+```
+
+To access or change a field of the JSON object `fruitDetailObject`, use one of the following:
+
+!!! note
+    If you define the JSON object using a `json` type variable, you need to cast it to a `map<json>` first.
+    
+```ballerina
+   fruitDetailObject["color"] = "orange"; 
+   
+   fruitDetailObject["name"];
```

Comment:
To access you can also use `fruitDetailObject.name` even when fruitDetailObject is not map<json>.